### ISSUE TITLE
LaTeX blocks munch on the following text

### ISSUE THREAD

**Issue by michaelcadilhac** [mattermost/mattermost-server#8683]
#### Summary
LaTeX blocks (introduced by triple backquotes) remove the text that immediately follows.

#### Steps to reproduce
Send:

    ```latex
    x```
    y

#### Expected behavior
This should display x, then y.

#### Observed behavior (that appears unintentional) 
y is removed.

#### Possible fixes
This line in [the markdown processor](https://github.com/mattermost/mattermost-webapp/blob/b849ec48bb89dda406275b8cecf7925261f14269/utils/markdown/renderer.jsx#L27) should be changed to:

     return `<div data-latex="${TextFormatting.escapeHtml(code)}"></div>`;

However, this is symptomatic of a bigger problem, I believe, since the current code does make sense.  I do not know where the mistake is introduced.  In particular, if LaTeX is disabled, the HTML code produced won't be

    <div data-latex="x"/>y

but

    <div data-latex="x">y</div>

**Comment by hmhealey** [comment]
Hi @michaelcadilhac,

This is actually because of a couple quirks with how Markdown handles code blocks. Markdown doesn't allow any other text when closing a code block, and it autocloses any unclosed code blocks at the end of the text, so
~~~
```latex
a```
b
~~~
renders a latex block containing
~~~
a```
b
~~~

You can see the same behaviour on GitHub
![image](https://user-images.githubusercontent.com/3277310/39428672-417ced40-4c56-11e8-8acf-c888f838bf4a.png)

**Comment by hmhealey** [comment]
That's really interesting. I've filed a Jira ticket to further investigate that behaviour [here](https://mattermost.atlassian.net/browse/MM-10403).

**Comment by hmhealey** [comment]
Oh, I should've read your report more thoroughly. The fix you suggested is exactly what I came across. It's due to https://github.com/aknuds1/html-to-react/issues/39

**Comment by lindy65** [comment]
Thanks @michaelcadilhac 

I'll close this issue off here for now as the PR has been merged into master and the fix will be in v4.10 releasing on May 16th.
